### PR TITLE
Fix ray stop

### PR DESCRIPTION
**Pull request by respecteverything**

### CODE REVIEW

File: pyzoo/zoo/ray/util/process.py [hunk 1]
```diff
@@ -125,7 +125,9 @@ def print_ray_remote_err_out(self):
     def clean_fn(self):
         import ray
```
Comment (zhichao-li) [inline pyzoo/zoo/ray/util/process.py]: Add a checking here.
```
if self.stopped:
    return
```

Comment (zhichao-li) [reply pyzoo/zoo/ray/util/process.py]: `self.stopped = True` at the init method?

File: pyzoo/test/zoo/ray/integration/test_multi.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# ray_ctx = RayContext(sc=sc, object_store_memory="1g")
+ray_ctx.init()
+actors = [TestRay.remote() for i in range(0, node_num)]
+print([ray.get(actor.hostname.remote()) for actor in actors])
+ray_ctx.stop()
+# sc.stop()
+print("------------------------first time done!--------------------")
+
+# repeat
+ray_ctx = RayContext(sc=sc, object_store_memory="1g")
```
Comment (zhichao-li) [inline pyzoo/test/zoo/ray/integration/test_multi.py]: remove this line

File: pyzoo/test/zoo/ray/integration/test_multi.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ray_ctx.init()
+actors = [TestRay.remote() for i in range(0, node_num)]
+print([ray.get(actor.hostname.remote()) for actor in actors])
+ray_ctx.stop()
+
+sc.stop()
+time.sleep(3)
+for process_info in ray_ctx.ray_processesMonitor.process_infos:
```
Comment (zhichao-li) [inline pyzoo/test/zoo/ray/integration/test_multi.py]: remove these checking as they are meaningless for yarn mode.

File: pyzoo/test/zoo/ray/integration/test_multi.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    hadoop_conf="/opt/work/hadoop-2.7.2/etc/hadoop/",
+    conda_name="rayexample",
+    num_executor=node_num,
+    executor_cores=28,
+    executor_memory="10g",
+    driver_memory="2g",
+    driver_cores=4,
+    extra_executor_memory_for_ray="30g")
+ray_ctx = RayContext(sc=sc, object_store_memory="2g")
+# sc = init_spark_on_local(cores=node_num)
```
Comment (zhichao-li) [inline pyzoo/test/zoo/ray/integration/test_multi.py]: remove this useless code.

File: pyzoo/test/zoo/ray/test_ray_multi.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            sc = init_spark_on_local(cores=node_num)
+            ray_ctx = RayContext(sc=sc, object_store_memory="1g")
+            ray_ctx.init()
+            actors = [TestRay.remote() for i in range(0, node_num)]
+            print([ray.get(actor.hostname.remote()) for actor in actors])
+            ray_ctx.stop()
+            time.sleep(3)
+            # repeat
+            print("-------------------first repeat begin!------------------")
+            ray_ctx = RayContext(sc=sc, object_store_memory="1g")
```
Comment (zhichao-li) [inline pyzoo/test/zoo/ray/test_ray_multi.py]: remove this line

File: pyzoo/test/zoo/ray/integration/test_multi.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+node_num = 4
+sc = init_spark_on_yarn(
+    hadoop_conf="/opt/work/hadoop-2.7.2/etc/hadoop/",
+    conda_name="rayexample",
+    num_executor=node_num,
+    executor_cores=28,
+    executor_memory="10g",
+    driver_memory="2g",
```
Comment (zhichao-li) [inline pyzoo/test/zoo/ray/integration/test_multi.py]: rename this file as well

**Comment by respecteverything** [comment]
PR Validation success (ignore the py27 test) : http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/2588/

**Pull request merged**